core/lib/types/src/block.rs

```diff
@@ -85,6 +88,43 @@ pub struct MiniblockHeader {
     pub virtual_blocks: u32,
 }
 
+/// Consensus-related L2 block (= miniblock) fields.
+#[derive(Debug, Clone)]
+pub struct ConsensusBlockFields {
+    /// Hash of the previous consensus block.
+    pub parent: validator::BlockHeaderHash,
+    /// Quorum certificate for the block.
+    pub justification: validator::CommitQC,
+}
+
+impl ProtoFmt for ConsensusBlockFields {
+    type Proto = crate::proto::ConsensusBlockFields;
+    fn read(r: &Self::Proto) -> anyhow::Result<Self> {
+        Ok(Self {
+            parent: read_required(&r.parent).context("parent")?,
+            justification: read_required(&r.justification).context("justification")?,
+        })
+    }
+    fn build(&self) -> Self::Proto {
+        Self::Proto {
+            parent: Some(self.parent.build()),
+            justification: Some(self.justification.build()),
+        }
+    }
+}
+
+impl Serialize for ConsensusBlockFields {
+    fn serialize<S: serde::Serializer>(&self, s: S) -> Result<S::Ok, S::Error> {
+        zksync_protobuf::serde::serialize(self, s)
+    }
+}
+
+impl<'de> Deserialize<'de> for ConsensusBlockFields {
+    fn deserialize<D: serde::Deserializer<'de>>(d: D) -> Result<Self, D::Error> {
+        zksync_protobuf::serde::deserialize(d)
+    }
+}
```

Comment:
Couldn't you use new macros in `zksync_protobuf::serde` to shorten this boilerplate? Or do you prefer explicitness?

Reply:
there is no new macros there now.